src/watertap_contrib/reflo/analysis/multiperiod/ltmed_vagmd_semibatch/MED_VAGMD_semibatch_class.py

```diff
@@ -0,0 +1,767 @@
+        # Take MD feed properties from last step mixer
+        (t1.fs.S2.MD_feed_state[0].temperature, t2.fs.vagmd.feed_props[0].temperature),
+        (t1.fs.S2.MD_feed_state[0].pressure, t2.fs.vagmd.feed_props[0].pressure),
+        (
+            t1.fs.S2.MD_feed_state[0].flow_mass_phase_comp["Liq", "TDS"],
+            t2.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "TDS"],
+        ),
+        (
+            t1.fs.S2.MD_feed_state[0].flow_mass_phase_comp["Liq", "H2O"],
+            t2.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "H2O"],
+        ),
+        # Inherit mixer properties from last step
+        (
+            t1.fs.S2.remained_liquid_state[0].temperature,
+            t2.fs.M1.remained_liquid_state[0].temperature,
+        ),
+        (
+            t1.fs.S2.remained_liquid_state[0].pressure,
+            t2.fs.M1.remained_liquid_state[0].pressure,
+        ),
+        (
+            t1.fs.S2.remained_liquid_state[0].flow_mass_phase_comp["Liq", "TDS"],
+            t2.fs.M1.remained_liquid_state[0].flow_mass_phase_comp["Liq", "TDS"],
+        ),
+        (
+            t1.fs.S2.remained_liquid_state[0].flow_mass_phase_comp["Liq", "H2O"],
+            t2.fs.M1.remained_liquid_state[0].flow_mass_phase_comp["Liq", "H2O"],
+        ),
+        # Mixer volume
+        (t1.fs.volume_in_tank, t2.fs.volume_in_tank_previous),
+    ]
+
+
+def unfix_dof(m):
+    """
+    This function unfixes a few degrees of freedom for optimization
+    Args:
+        m: object containing the integrated nuclear plant flowsheet
+    Returns:
+        None
+    """
+
+    m.fs.vagmd.feed_props[0].temperature.unfix()
+    m.fs.vagmd.feed_props[0].pressure.unfix()
+    m.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "H2O"].unfix()
+    m.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "TDS"].unfix()
+
+    m.fs.M1.remained_liquid_state[0].temperature.unfix()
+    m.fs.M1.remained_liquid_state[0].pressure.unfix()
+    m.fs.M1.remained_liquid_state[0].flow_mass_phase_comp["Liq", "H2O"].unfix()
+    m.fs.M1.remained_liquid_state[0].flow_mass_phase_comp["Liq", "TDS"].unfix()
+
+    m.fs.volume_in_tank_previous.unfix()
+
+    return
+
+
+@declare_process_block_class("MEDVAGMDsemibatch")
+class MEDVAGMDsemibatchData(UnitModelBlockData):
+
+    CONFIG = ConfigBlock()
+
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([False]),
+            default=False,
+            description="Dynamic model flag - must be False",
+            doc="""Indicates whether this model will be dynamic or not,
+    **default** = False.""",
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Indicates whether holdup terms should be constructed or not.
+    **default** - False.""",
+        ),
+    )
+    CONFIG.declare(
+        "model_input",
```

Comment:
Any value in having these as separate CONFIG arguments?

Reply:
In the `multiperiod` class the flowsheet options to each period are provided as a dictionary, where the values should be numbers but not `pyomo.Var` object. So I just put these must-have inputs in the CONFIG instead of creating additional `Var` objects, fixing their values and passing to the fs options. Once the multiperiod class is constructed, we can always unfix these variables for optimization or other purposes. Hope it makes sense!